File: app/src/main/java/com/waray/spendhound/PayerGroup.java
package com.waray.spendhound;

import java.util.List;

public class PayerGroup {
    private String groupId;
    private String groupName;
    private List<String> members;           // Now stores UIDs (was usernames)
    private String createdBy;
    private List<String> memberDisplayNames; // Display names for UI

    public PayerGroup() {
        // Default constructor required for Firebase
    }

    public PayerGroup(String groupId, String groupName, List<String> members, String createdBy) {
        this.groupId = groupId;
        this.groupName = groupName;
        this.members = members;
        this.createdBy = createdBy;
    }

    // New constructor with display names
    public PayerGroup(String groupId, String groupName, List<String> members, String createdBy, List<String> memberDisplayNames) {
        this.groupId = groupId;
        this.groupName = groupName;
        this.members = members;
        this.createdBy = createdBy;
        this.memberDisplayNames = memberDisplayNames;
    }

    public String getGroupId() {
        return groupId;
    }

    public void setGroupId(String groupId) {
        this.groupId = groupId;
    }

    public String getGroupName() {
        return groupName;
    }

    public void setGroupName(String groupName) {
        this.groupName = groupName;
    }

    public List<String> getMembers() {
        return members;
    }

    public void setMembers(List<String> members) {
        this.members = members;
    }

    public String getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }

    public List<String> getMemberDisplayNames() {
        return memberDisplayNames;
    }

    public void setMemberDisplayNames(List<String> memberDisplayNames) {
        this.memberDisplayNames = memberDisplayNames;
    }
}

File: app/src/main/java/com/waray/spendhound/PayerGroup.kt
package com.waray.spendhound

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

@Serializable
data class PayerGroup(
    @SerialName("group_id")
    var groupId: Long? = null,
    @SerialName("group_name")
    var groupName: String? = null,
    @SerialName("createdby_id")
    var createdBy: Long? = null,
    @SerialName("created_at")
    var createdAt: String? = null,
    @Transient
    var memberDisplayNames: MutableList<String?>? = null
)

@Serializable
data class GroupMember(
    @SerialName("id")
    val id: Long? = null,
    @SerialName("group_id")
    val groupId: Long? = null,
    @SerialName("user_id")
    val userId: Long? = null,
    @SerialName("created_at")
    val createdAt: String? = null
)
